daemon/network.go

```diff
@@ -373,12 +373,12 @@ func (daemon *Daemon) createNetwork(cfg *config.Config, create types.NetworkCrea
 		nwOptions = append(nwOptions, libnetwork.NetworkOptionLBEndpoint(nodeIP))
 	}
 
+	if driver == "overlay" && !daemon.cluster.IsManager() {
+		return nil, errdefs.Forbidden(errors.New("This node is not a swarm manager. Use \"docker swarm init\" or \"docker swarm join\" to connect this node to swarm and try again."))
+	}
```

Comment:
I see there's a bunch of code / logic happening before this. Would it be possible to move this check more to the start of this function? Maybe even before the `nw, err := daemon.GetNetworkByName(create.Name)` line (_technically_ that would mean we may have to move the code that gets the default driver as well, but that default is not user-configurable, so would never be `overlay`).